Update User Guide - Alternative layouts for the TOC (#10131)
geosolutions-it/MapStore2: docs/user-guide/toc.md

* Hide all groups and layers present in the TOC, enabling the **Hide all child nodes** through the <img src="../img/button/hide-all-child-nodes.jpg" class="ms-docbutton"/> button.

* Collapse all groups present in the TOC, enabling the **Collapse all child nodes** through the <img src="../img/button/collapse-all-child-nodes.jpg" class="ms-docbutton"/> button.

* Expand all groups present in the TOC, enabling the  **Expand all child nodes** through the <img src="../img/button/expand-all-child-nodes.jpg" class="ms-docbutton"/> button. Each layer's legends are also expanded.

* Add a group under all the others groups present on TOC, through the <img src="../img/button/add-group.jpg" class="ms-docbutton"/> button.

* Zoom the map to TOC's layers extent, through the <img src="../img/button/zoom-extent.jpg" class="ms-docbutton"/> button.

### Search for layers

With the TOC it is also possible to perform a search between the added layers. This operation can be done simply by typing the name (o part of it) of the layer in the search bar:

<img src="../img/toc/search.jpg" class="ms-docimage"  style="max-width:300px;"/>

### Choose layers and groups position

> The user can decide to change the position of a *Layer* within the TOC by clicking on the layer <img src="../img/button/drag-drop-button.jpg" class="ms-docbutton"/> and drag and drop it. <br>

Reviewer: ```suggestion
The user can decide to change the position of a *Layer* within the TOC by clicking on the layer's drag and drop icon <img src="../img/button/drag-drop-button.jpg" class="ms-docbutton"/>. <br>
```

> It is possible to change layers position inside the same group, but also moving them between different groups. With this operation the position of the layer also changes on the map. <br>

Reviewer: ```suggestion
It is possible to change the position of a layer within its group or move it into a different groups. Once this is done, also the order of the layer in the map changes accordingly. <br>
```

> Changing layers position with the drag and drop, for example, it can display like the following:

Reviewer: ```suggestion
Below is an example of changing the layer position through the drag and drop function.
```

<video class="ms-docimage" controls><source src="../img/toc/ded-layers.mp4"/></video>

The same operation can also be done with *Groups* and *Sub-groups*, no matter their layer, can be nested inside other groups and sub-groups, or can be separated from their parent-layer to create new main groups.

<video class="ms-docimage" controls><source src="../img/toc/ded-groups.mp4"/></video>

Layers position can also be determined through the **Selected layer settings** button <img src="../img/button/properties.jpg" class="ms-docbutton"/> available in the toolbar that appears once a layer is selected. This button opens a panel where the user can choose the destination group (or subgroup):

<img src="../img/toc/settings-group.jpg" class="ms-docimage"  style="max-width:350px;"/>

### Display options in panel

> Directly from the TOC panel, the user can set different types of display options. In particular, for layers, it is possible to:

Reviewer: ```suggestion
Directly from the TOC UI, the user can access different kind of display options. In particular, for layers, it is possible to:
```

<img src="../img/toc/layer-legend.jpg" class="ms-docimage"/>

> * Enable/disable the layer visibility by toggle the check box to the left of the layer name <img src="../img/button/check-box.jpg" class="ms-docbutton"/>

Reviewer: ```suggestion
* Enable/disable the layer visibility by using the check box to the left side of the layer leaf<img src="../img/button/check-box.jpg" class="ms-docbutton"/>
```

* Expand <img src="../img/button/expand-all-child-nodes.jpg" class="ms-docbutton"/> or collapse <img src="../img/button/collapse-all-child-nodes.jpg" class="ms-docbutton"/> the layer legend.

* Control the layer transparency in map by scrolling the opacity slider

For the groups the user can:

<img src="../img/toc/group.jpg" class="ms-docimage" style="max-width:500px;"/>

> * Expand <img src="../img/button/expand-all-child-nodes.jpg" class="ms-docbutton"/> or collapse <img src="../img/button/collapse-all-child-nodes.jpg" class="ms-docbutton"/> the list of layers or subgroups nested inside it.

Reviewer: ```suggestion
* Expand <img src="../img/button/expand-all-child-nodes.jpg" class="ms-docbutton"/> or collapse <img src="../img/button/collapse-all-child-nodes.jpg" class="ms-docbutton"/> layers or groups inside.
```

> * Enable/disable the group visibility by toggle the check box to the left of the group name <img src="../img/button/check-box.jpg" class="ms-docbutton"/>

Reviewer: ```suggestion
* Enable/disable the group visibility by using the check box to the left of the group leaf <img src="../img/button/check-box.jpg" class="ms-docbutton"/>
```

!!! note

>     When the user switch off the visibility of a group, the visibility of all layers and subgroups nested within it will not change.

Reviewer: ```suggestion
   When the user turns off the visibility of a group, the visibility in map of all layers and groups inside it changes accordingly but their original visibility state in TOC remain untouched; simply all nested elements are grayed out to indicate they are not visible on map and other functions (e.g. via context menu) remain available. 
```

## Group Settings and Toolbar

Once a group is selected the following toolbar appears and the user can:

<img src="../img/toc/group-toolbar.jpg" class="ms-docimage"/>

* **Add layer to selected group** through the <img src="../img/button/add_layer_button.jpg" class="ms-docbutton"/> button (it is possible to add one or more layers to the group)

> * **Add sub group to the selected group** through the <img src="../img/button/add_group_button.jpg" class="ms-docbutton"/> button (it is possible to add one or more subgroups to the selected group)

Reviewer: ```suggestion
* **Add subgroup to the selected group** through the <img src="../img/button/add_group_button.jpg" class="ms-docbutton"/> button (it is possible to add one or more subgroups to the selected group)
```

> * **Zoom to selected layers extent** in order to zoom the map to all layers belonging to the group, through the <img src="../img/button/zoom-layer.jpg" class="ms-docbutton"/> button

Reviewer: ```suggestion
* **Zoom to selected layers extent** in order to zoom the map to the extent covering all layers belonging to the given group, through the <img src="../img/button/zoom-layer.jpg" class="ms-docbutton"/> button
```

* Open the **Group settings** through the <img src="../img/button/properties.jpg" class="ms-docbutton"/> button: the *Settings* panel opens and the user can:<br>
  **-** Change the **Title**<br>

>   **-** Set the translation of the group title by opening the **Localize Text** popup through the <img src="../img/button/localize_button.jpg" class="ms-docbutton"/> button. This way the language of the title changes according to the  current language setting in [MapStore](https://mapstore.geosolutionsgroup.com/mapstore/#/)<br>

Reviewer: ```suggestion
  **-** Set the translation of the group title by opening the **Localize Text** popup through the <img src="../img/button/localize_button.jpg" class="ms-docbutton"/> button. This way the language of the title changes according to the  current [MapStore](https://mapstore.geosolutionsgroup.com/mapstore/#/) language <br>
```

  **-** Edit the group's **Description**<br>

>   **-** Configure the **Tooltip** that appears moving the cursor over the group's item in TOC. In this case the user can decide that the *Title*, the *Description*, both or nothing will be displayed. Moreover you can set the *Placement* of the tooltip, choosing between *Top*, *Right* or *Bottom*.

Reviewer: ```suggestion
  **-** Configure the **Tooltip** that appears moving the cursor over the group's item in TOC. In this case the user can decide that the *Title*, the *Description*, both or nothing will be displayed. Moreover you can set the *Placement* of the tooltip, choosing between *Top*, *Right* or *Bottom*. In all circumstances the title is fully visible in TOC the tooltip doesn't appear.
```

<img src="../img/toc/group-settings-panel.jpg" class="ms-docimage"/>

* **Remove selected group** and its content through the <img src="../img/button/delete.jpg" class="ms-docbutton"/> button

> Click-right on a group, the user can modify some group display properties such as:

Reviewer: ```suggestion
By right clicking on a group, the user can change some properties of the group such as:
```

<video class="ms-docimage" controls><source src="../img/toc/group-settings-panel.mp4"/></video>

* Show all groups and layers present in the selected group, enabling the **Show all child nodes** through the <img src="../img/button/show-all-child-nodes.jpg" class="ms-docbutton"/> button.

* Hide all groups and layers present in the selected group, enabling the **Hide all child nodes** through the <img src="../img/button/hide-all-child-nodes.jpg" class="ms-docbutton"/> button.

> * Activate mutually exclusive visibility of groups and layers present in the selected group, enabling the **Activate mutually exclusive visibility of child nodes** through the <img src="../img/button/mutually-exclusive-visibility.jpg" class="ms-docbutton"/> button

Reviewer: ```suggestion
* Activate a mutually exclusive visibility of groups and layers inside the selected group, enabling the **Activate mutually exclusive visibility of child nodes** through the <img src="../img/button/mutually-exclusive-visibility.jpg" class="ms-docbutton"/> button. In this way, only one layer or subgroup within it at a time can be visible on map.
```

* Collapse all groups present in the selected group, enabling the **Collapse all child nodes** through the <img src="../img/button/collapse-all-child-nodes.jpg" class="ms-docbutton"/> button.

> * Expand all groups present in the selected group, enabling the  **Expand all child nodes** through the <img src="../img/button/expand-all-child-nodes.jpg" class="ms-docbutton"/> button. Each layer's legends are also expanded.

Reviewer: ```suggestion
* Expand all groups present in the selected group, enabling the  **Expand all child nodes** through the <img src="../img/button/expand-all-child-nodes.jpg" class="ms-docbutton"/> button. Each layer legend is also expanded.
```

* Add a subgroup under the selected group, through the <img src="../img/button/add-group.jpg" class="ms-docbutton"/> button.

> * Zoom the map to the selected group's layers extent, through the <img src="../img/button/zoom-extent.jpg" class="ms-docbutton"/> button.

Reviewer: ```suggestion
* Zoom the map to the selected group layers extent, through the <img src="../img/button/zoom-extent.jpg" class="ms-docbutton"/> button.
```

!!! note
    The **Metadata Tool** is not configured by default in [MapStore](https://mapstore.geosolutionsgroup.com/mapstore/#/). A complete documentation to configure it is available as part of the [TOC Plugins documentation](https://mapstore.geosolutionsgroup.com/mapstore/docs/api/plugins#plugins.TOC) (see *metadataOptions*). Once the **Metadata Tool** has been configured, MapStore is able to load the layer metadata from the remote CSW service and parse it to be presented to the user according to the provided plugin configuration. This functionality automatically works in case of WMS layers coming from a CSW catalog source, while for layers coming directly from a WMS catalog source the [Metadata Link](https://docs.geoserver.org/latest/en/user/data/webadmin/layers.html#basic-info) must be present in the WMS Layer GetCapabilities.

> Click-right on a layer, the user can modify some layer display properties such as:

Reviewer: ```suggestion
By right clicking on a layer, the user can manage some layer properties such as:
```
